examples/vision/swin_transformers.py

```diff
@@ -0,0 +1,557 @@
+representation is computed with shifted windows. The shifted windowing scheme 
+brings greater efficiency by limiting self-attention computation to 
+non-overlapping local windows while also allowing for cross-window connection. 
+This architecture also has the flexibility to model at various scales and has 
+linear computational complexity with respect to image size.
+
+This example requires TensorFlow 2.5 or higheras well as TensorFlow Addons, 
+which can be installed using the following commands:
+"""
+
+"""shell
+pip install -U tensorflow-addons
+"""
+
+"""
+## Setup
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+import tensorflow as tf
+import tensorflow_addons as tfa
+from tensorflow import keras
+from tensorflow.keras import layers
+from tensorflow.python.keras.layers.pooling import GlobalAveragePooling1D
+
+"""
+## Prepare the data
+
+We will now load the CIFAR-100 dataset through 
+[tf.keras.datasets](https://www.tensorflow.org/api_docs/python/tf/keras/datasets)
+, normalize the images and convert label integers to matrices.
+"""
+
+num_classes = 100
+input_shape = (32, 32, 3)
+
+(x_train, y_train), (x_test, y_test) = keras.datasets.cifar100.load_data()
+x_train, x_test = x_train / 255.0, x_test / 255.0
+y_train = keras.utils.to_categorical(y_train, num_classes)
+y_test = keras.utils.to_categorical(y_test, num_classes)
+print(f"x_train shape: {x_train.shape} - y_train shape: {y_train.shape}")
+print(f"x_test shape: {x_test.shape} - y_test shape: {y_test.shape}")
+
+plt.figure(figsize=(10, 10))
+for i in range(25):
+    plt.subplot(5, 5, i + 1)
+    plt.xticks([])
+    plt.yticks([])
+    plt.grid(False)
+    plt.imshow(x_train[i])
+plt.show()
+
+"""
+## Configure the hyperparameters
+
+In order to use each pixel as an individual input, you can set `patch_size` to (1, 1).
+We take inspirration from the original paper where they share the experimental settings
+for training on ImageNet-1K, keeping quite some of the settings same for this example.
+"""
+
+patch_size = (2, 2)  # 2-by-2 sized patches
+dropout_rate = 0.03  # Dropout rate
+num_heads = 8  # Attention heads
+embed_dim = 64  # Embedded dimensions
+num_mlp = 256  # MLP nodes
+qkv_bias = True  # Convert embedded patches to query, key, and values with a learnable additive value
+window_size = 2  # Size of attention window
+shift_size = 1  # Size of shifting
+image_dimension = 32
+
+num_patch_x = input_shape[0] // patch_size[0]
+num_patch_y = input_shape[1] // patch_size[1]
+
+learning_rate = 1e-3
+batch_size = 128
+num_epochs = 40
+validation_split = 0.1
+weight_decay = 0.0001
+label_smoothing = 0.1
+
+"""
+## Helper Functions
+
+We will now create two helper functions which can help us get a sequence of 
+patches from the image, allow us to merge patches to spatial frames and dropout.
+"""
+
+
+def window_partition(x, window_size):
+    _, height, width, channels = x.shape
+    patch_num_H = height // window_size
```

Comment:
Style: use lowercase (snake case) variable names everywhere

Reply:
Thanks for pointing this out, might  have missed this one in an earlier commit